src/thermo/ThermoFactory.cpp

```diff
@@ -451,6 +451,12 @@ void setupPhase(ThermoPhase& thermo, AnyMap& phaseNode, const AnyMap& rootNode)
         phaseNode["__file__"] = rootNode["__file__"];
     }
 
+    if (phaseNode.hasKey("deprecated")) {
+        string msg = phaseNode["deprecated"].asString();
+        string method = fmt::format("Phase deprecation: '{}'", phaseNode["name"].asString());
+        warn_deprecated(method, msg);
```

Comment:
Up to this point, deprecation warnings have been used for deprecated methods only (the output should also have consistent formatting which is similar to errors). After #741, there is now `warn_user`

Reply:
@ischoegl IMO using `warn_deprecated` is fine here. It gives a nice control over whether the warnings should be fatal (e.g., making sure we don't use deprecated files/phases in the test suite) and gives users the ability to silence them.

@speth I added the filename and the phase name. If only the filename is added, then additional warnings from other phases in the same file would be suppressed.

Reply:
@bryanwweber ... no problem, just wanted to point this out. The other observation remains: the `method` string in its present form is not consistent with usage implemented elsewhere (it would be simply `setupPhase`, with the rest of the message being merged with `msg`)

Reply:
@ischoegl yes, that would be consistent with the documentation and usage elsewhere but if the `method` is not unique, the message will be suppressed when it shouldn't be, e.g., multiple deprecated phases in the same file. Is there a particular solution to this you had in mind? I could update the documentation or the argument name

Reply:
I don’t have a particular solution, but would maintain the current convention of error/warning signatures whatever the final solution may be. To that end, I’d still suggest `warn_user`, as the deprecation involves user input. Presumably, the code base does not change, so I’m not sure that this situation warrants a redefinition of `warn_deprecated`. Just 2 cents, and I won’t press this further.

Reply:
I think `warn_deprecated` is still the right function to call here, since we are warning about a deprecation. It's just that the argument name `method` is not quite generic enough -- it should just be "the name of the deprecated thing", so in this case, "*foo*.yaml/*bar*" makes sense.

For `warn_user`, the purpose of the `method` argument is to tell you where the warning is coming from, but in this case that's not really what's important.